test/functional/feature_assumeutxo.py

```diff
@@ -68,23 +68,35 @@ def setup_network(self):
 
     def test_invalid_snapshot_scenarios(self, valid_snapshot_path):
         self.log.info("Test different scenarios of loading invalid snapshot files")
-        self.log.info("  - snapshot file refering to a block that is not in the assumeutxo parameters")
         with open(valid_snapshot_path, 'rb') as f:
             valid_snapshot_contents = f.read()
+        bad_snapshot_path = valid_snapshot_path + '.mod'
 
+        self.log.info("  - snapshot file refering to a block that is not in the assumeutxo parameters")
         # we can only test this with a block that is already known, as otherwise the `loadtxoutset` RPC
         # would time out (waiting to see the hash in the headers chain), rather than error immediately
         bad_snapshot_height = SNAPSHOT_BASE_HEIGHT - 1
-        bad_snapshot_path = valid_snapshot_path + '.mod'
         with open(bad_snapshot_path, 'wb') as f:
             bad_snapshot_block_hash = self.nodes[0].getblockhash(bad_snapshot_height)
             # block hash of the snapshot base is stored right at the start (first 32 bytes)
             f.write(bytes.fromhex(bad_snapshot_block_hash)[::-1] + valid_snapshot_contents[32:])
 
         expected_log = f"assumeutxo height in snapshot metadata not recognized ({bad_snapshot_height}) - refusing to load snapshot"
-        with self.nodes[1].assert_debug_log(expected_log):
+        with self.nodes[1].assert_debug_log([expected_log]):
```

Comment:
fafde92f84fb7c245bc3c1cd946a32c891861e5e: I verified this test indeed would pass regardless of `expected_log`. Wen type safety?